designs/2019-linter-options/README.md

```diff
@@ -0,0 +1,126 @@
+- Start Date: 2019-05-12
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# The `linterOptions` property in Config Files
+
+## Summary
+
+This RFC adds `linterOptions` property to config files. People can use config files (including shareable configs!) instead of CLI options in order to configure some linter behavior.
+
+## Motivation
+
+We cannot configure some linter's behavior with config files, especially, shareable configs. It's convenient if we can configure those in shareable configs.
+
+## Detailed Design
+
+This adds `linterOptions` property to config files with three properties.
+
+```js
+// .eslintrc.js
+module.exports = {
+    linterOptions: {
+        allowInlineConfig: true,              // Corresponds to --no-inline-config / `options.allowInlineConfig`
+        reportUnusedDisableDirectives: "off", // Corresponds to --report-unused-disable-directives / `options.reportUnusedDisableDirectives`
```

Comment:
Personally, I'm still not sure it's a good idea to allow `reportUnusedDisableDirectives` to be specified in a shareable config for the reasons discussed in https://github.com/eslint/eslint/issues/9382.

It also seems like it could also be confusing if a shareable config specified `allowInlineConfig: false`. (For example, it could be very difficult for a user to figure out why their `// eslint-disable` comments are being ignored.)

In general, I'm fine with the idea of adding these to config files, but it doesn't seem ideal to allow them to be added to shareable configs.

Reply:
> Personally, I'm still not sure it's a good idea to allow reportUnusedDisableDirectives to be specified in a shareable config for the reasons discussed in eslint/eslint#9382.

I have used `eslint-comments/no-unused-disable` rule in my shareable config and my company config while a long time. It didn't seem a problem to us. Anyway, selectable is good to me.

Or `"warn"` level doesn't fail CI builds, so I'm OK if we make `linterOptions.reportUnusedDisableDirectives` boolean and it means `"off"` or `"warn"`.

> It also seems like it could also be confusing if a shareable config specified allowInlineConfig: false. (For example, it could be very difficult for a user to figure out why their // eslint-disable comments are being ignored.)

I agree that that situation is confusable.
I don't feel strongly about adding `allowInlineConfig`.